src/Maestro/DependencyUpdater/DependencyUpdater.cs

```diff
@@ -330,19 +334,22 @@ where sub.Enabled
 
     private async Task UpdateSubscriptionAsync(Guid subscriptionId, int buildId)
     {
-        using (_operations.BeginOperation(
-                   "Updating subscription '{subscriptionId}' with build '{buildId}'",
-                   subscriptionId,
-                   buildId))
+        if (_subscriptionIdGenerator.ShouldTriggerSubscription(subscriptionId))
```

Comment:
We should at least log here otherwise we will be triggering subscriptions and scratching our heads why nothing happened.

Also I would invert the if statement, log and return early from here.